app/src/pages/[resourceType]/[id].tsx

```diff
@@ -211,6 +245,21 @@ export default function ResourceIDPage({ jsonData }: InferGetServerSidePropsType
               </ScrollArea.Autosize>
             </Tabs.Panel>
           )}
+          {dataRequirements?.relatedArtifact && (
+            <Tabs.Panel value="dependencies">
+              <Space h="md" />
+              <Text c="dimmed">
+                Number of Dependencies:<b> {dataRequirements.relatedArtifact.length} </b>
+              </Text>
+              <Space h="md" />
+              <ScrollArea.Autosize mah={height * 0.8} type="hover">
+                {dataRequirements.relatedArtifact.map((relatedArtifact, index) => (
+                  <Dependency key={index} relatedArtifact={relatedArtifact} />
```

Comment:
This is super niche, but it is actually not recommended by React to use the index of iteration as the key on a DOM element. The docs list a pitfall for this [here](https://react.dev/learn/rendering-lists#why-does-react-need-keys). In short, it could cause weird bugs or performance issues in some cases.

Instead, we should make or use a string that is unique for each dependency. I think it's safe to assume that for a given `relatedArtifact`, the `.resource` attribute will be unique. So you can probably use `relatedArtifact.resource` as the key.

If this wasn't the case, the following strategy also works:

```tsx
                {dataRequirements.relatedArtifact.map((relatedArtifact, index) => {
                  // Using the index outside of the actual `key` prop assignment is an okay workaround
                  const key = `dependency-${index}`;
                  return <Dependency key={key} relatedArtifact={relatedArtifact} />;
                })}
```